### ISSUE TITLE
solo5.h, solo5.o should export only namespaced public interfaces

### ISSUE THREAD

**Issue by mato** [djwillia/solo5#23]
solo5.o currently exports all symbols defined inside the kernel. solo5.h likewise defines "everything needed by the kernel to build" rather than just the public interfaces. To enforce clear layering (needed by #19), only public symbols and interfaces should be exported.

**Comment by mato** [comment]
Proposed list of public interfaces so far:
````
/* Network */
int solo5_net_write_sync(uint8_t *data, int n);
int solo5_net_read_sync(uint8_t *data, int *n);
char *solo5_net_mac_str(void);

/* Block */
int solo5_blk_write_sync(uint64_t sec, uint8_t *data, int n);
int solo5_blk_read_sync(uint64_t sec, uint8_t *data, int *n);
int solo5_blk_sector_size(void);
uint64_t solo5_blk_sectors(void);
int solo5_blk_rw(void);

/* Console */
int solo5_console_write(const char *buf, size_t n);

/* Exit */
void solo5_exit(void) __attribute__((noreturn));

/* Memory allocation */
void *solo5_malloc(size_t);
void solo5_free(size_t);
void *solo5_calloc(size_t, size_t);
void *solo5_realloc(void *, size_t);

/* Time */
uint64_t solo5_clock_monotonic(void);
uint64_t solo5_clock_wall(void); /* Not yet, but we will need a wall clock */
````
I deliberately do not want to export any "POSIXy" or libc interfaces directly, to discourage the notion that Solo5 supports such interfaces. A consequence of this is that some libc interfaces which are needed by the OCaml runtime (printf family, string.h) and also used internally in Solo5 will be duplicated in the solo5-ocaml-runtime layer (see #19). This is fine, the amount of code is insignificant (tens of kB).

**Comment by mato** [comment]
Fixed by #24
